### DISCUSSION TITLE
Feature Request: Option to have "iterations" in projects understand/exclude weekends and holidays

### DISCUSSION THREAD

**Discussion by tektrip-biggles** [community/community#57824] [Projects and Issues]
### Select Topic Area

Product Feedback

### Select Feature Area

- [ ] Issues (The feature that allows you to track ideas, feedback, tasks, or bugs for work on GitHub. Not for any issue you are having).
- [X] Projects (For planning and tracking work on GitHub - not any open source or other type of project).

### Body

I'm a solo dev using GitHub Projects to manage my workload on a particular project into two-week "sprints". These should generally start on a Monday and end on the following Friday, comprising of 10 days in total. If there is a (UK) bank holiday (or if I'm taking time off for some other reason) then the sprint will be shorter. While it's possible to have a "break" _between_ sprints/iterations in my project, it's somewhat cumbersome to have to add these in manually every time and it doesn't seem to be possible to manually add manual weekend breaks in the _middle_ of a sprint/iteration. As such, GitHub Projects reports each of my 10-day sprints as being 12 days long, which is not an accurate reflection of the time that will actually be spent doing work on the project. In a previous company, there was even a strict 4-day week policy, which I understand is becoming more popular generally.

Perhaps this is a small thing & many organisations will be perfectly happy just pretending that each sprint includes the weekends. However, in the interests of supporting those organisations that _do_ value encouraging good mental health and work-life balance,  as well as more accurately assessing project timescales more generally, it would be great to make it easier to have project boards more accurately reflect this in terms of reporting how many _actual_ work days are going to be part of an iteration.

UI-wise, I'm imagining an optional section of the "iteration settings" which allows one to specify which days of the week to normally consider to be working days (i.e. a checkbox for each) as well as the option to override that in specific instances (i.e. to mark national holidays as non-working days, or perhaps even to mark one-off working days on a weekend in exceptional circumstances). With these settings enabled, iterations would then count how many _working_ days there are between their start & end dates, so it would be unnecessary to add "breaks" manually between or in the middle of iterations. 

It may also be helpful to be able to display info about each iteration in terms of "first/last working days" at the top of the "current iteration" / "next iteration" views (e.g. "Sprint 1: Monday 5th June - Friday 16th June (10 days)") even if "under the hood" that iteration ran, say, "Sunday 4th - Saturday 17th". An extra-nice feature would be to have progress bars for the "current sprint" showing that time-wise the sprint is 50% done or something.

To make that last feature _even more_ useful, it would be good to be able to also specify what the "normal working hours" are (e.g. 9am - 6pm UK time) so that the project could know that I'm, say, 40% through the current day, or in order to calculate how many _working hours_ are available in a particular iteration.

I totally get that these features may not actually be useful to every team, especially highly distributed ones working across multiple time zones and cultures, but it feels like something that may be worth discussing at the very least?

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by Gustavonn07** [comment]
I agree with your ideas it would be very good to make it more dynamic.  To propose these ideas and discuss their feasibility, it would be best to reach out to GitHub directly through their official channels, such as their support forums or feedback mechanisms. This will allow you to share your suggestions directly with the GitHub team, who can consider them for future updates and improvements to GitHub Projects.

support: https://support.github.com/

**Comment by tektrip-biggles** [comment]
Ah, I thought that this _was_ the most appropriate "official channel" for reaching the GitHub team! In fact, if I click on the link you sent, then "Product Feedback" it takes me to these same discussion boards. Perhaps I'm missing something though? Is there a more appropriate option than "Product Feedback" I should look for?

**Reply by Gustavonn07** [reply]
hmmm I think the way now would be to wait for some platform administrator to see it.
 I gave it an up to try to help to have more visibility.

**Comment by austinpray-mixpanel** [comment]
I expected the weekends to at least be visually distinct from workdays

![board_blocked_weekends](https://github.com/community/community/assets/71290498/3a036d12-e4f8-4ee0-9c0c-3dd6fb44179c)

I would also expect "break" cycles to be visually differentiated 

![image](https://github.com/community/community/assets/71290498/7f9a206d-9e2c-4803-b8d4-0cb105ed4265)
